packages/riipen-ui/src/components/Typography.test.jsx

```diff
@@ -0,0 +1,289 @@
+import { mount } from "enzyme";
+import toJson from "enzyme-to-json";
+import React from "react";
+
+import Typography from "./Typography";
+// import { variantMapping } from "./Typography";
+
+describe("<Typography>", () => {
+  it("renders without errors", () => {
+    let error;
+
+    try {
+      mount(<Typography />);
+    } catch (theError) {
+      error = theError;
+    }
+
+    expect(error).toEqual(undefined);
+  });
+
+  it("renders correct snapshot", () => {
+    const wrapper = mount(<Typography />);
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  describe("default props", () => {
+    it("sets correct default props", () => {
+      const defaultProps = new Typography().type.defaultProps;
+
+      const wrapper = mount(<Typography />);
+
+      const component = wrapper.find("Typography");
+      expect(component.props().color).toEqual(defaultProps.color);
+      expect(component.props().display).toEqual(defaultProps.display);
+      expect(component.props().gutter).toEqual(defaultProps.gutter);
+      expect(component.props().textAlign).toEqual(defaultProps.textAlign);
+      expect(component.props().variant).toEqual(defaultProps.variant);
+    });
+  });
+
+  describe("children prop", () => {
+    it("renders given children", () => {
+      const child = <span>this is a child</span>;
+
+      const wrapper = mount(<Typography>{child}</Typography>);
+
+      expect(wrapper.find("Typography").contains(child)).toEqual(true);
+    });
+  });
+
+  describe("classes prop", () => {
+    it("applies classes to the root node of Typography", () => {
+      const classes = ["classOne"];
+
+      const wrapper = mount(<Typography classes={classes} />);
+
+      expect(
+        wrapper
+          .find("Typography")
+          .childAt(0)
+          .hasClass(classes[0])
+      ).toEqual(true);
+    });
+
+    it("appends higher order values to classes prop with withClasses decorator", () => {
+      const classes = ["riipen", "riipen-typography"];
+
+      const wrapper = mount(<Typography />);
+
+      expect(
+        wrapper
+          .find("Typography")
+          .props()
+          .classes.sort()
+      ).toEqual(classes.sort());
+    });
+
+    it("gives an error when classes are provided as one string", () => {
```

Comment:
Is this test necessary since we stopped using the test where two classes are added in the array?

Reply:
Ahh it's because html classes are commonly added as a string, like 
`<div class="one two three"></div>`.

Reply:
I mean, we _probably_ don't need this test. This test is actually testing to make sure `PropTypes` are working.

1. We have a prop of `classes` that has to be an array
2. We are trying to pass a string instead of array
3. We expect and error

This is more so testing React's internals rather than any code we have written.